### ISSUE TITLE
Gratuitous use of  'auto' keyword

### ISSUE THREAD

**Issue by bmerry** [HandsOnOpenCL/Lecture-Slides#25]
When the slides create a kernel function object, they say

``` c++
auto vadd = make_kernel<Buffer, Buffer, Buffer, int>(program, "vadd");
```

which is an unnecessarily roundabout way to say

``` c++
make_kernel<Buffer, Buffer, Buffer, int> vadd(program, "vadd");
```

and also makes it appear that the bindings depend on C++11.

**Comment by tomdeakin** [comment]
Ah, I didn't know that make_kernel could be used as a constructor, cheers.

**Comment by tomdeakin** [comment]
Fixed in v1.2 of the slides and code.
